Framework/Algorithms/src/He3TubeEfficiency.cpp

```diff
@@ -147,60 +145,63 @@ void He3TubeEfficiency::exec() {
   PARALLEL_CHECK_INTERUPT_REGION
 
   this->logErrors();
-  this->setProperty("OutputWorkspace", this->outputWS);
+  this->setProperty("OutputWorkspace", m_outputWS);
 }
 
 /**
- * Corrects a spectra for the detector efficiency calculated from detector
- * information. Gets the detector information and uses this to calculate its
- * efficiency
- *  @param spectraIndex :: index of the spectrum to get the efficiency for
- *  @throw invalid_argument if the shape of a detector is isn't a cylinder
- *  aligned along one axis
- *  @throw NotFoundError if the detector or its gas pressure or wall thickness
- *  were not found
- */
+* Corrects a spectra for the detector efficiency calculated from detector
+* information. Gets the detector information and uses this to calculate its
+* efficiency
+*  @param spectraIndex :: index of the spectrum to get the efficiency for
+*  @throw invalid_argument if the shape of a detector is isn't a cylinder
+*  aligned along one axis
+*  @throw NotFoundError if the detector or its gas pressure or wall thickness
+*  were not found
+*/
 void He3TubeEfficiency::correctForEfficiency(std::size_t spectraIndex) {
-  Geometry::IDetector_const_sptr det = this->inputWS->getDetector(spectraIndex);
+  Geometry::IDetector_const_sptr det = m_inputWS->getDetector(spectraIndex);
   if (det->isMonitor() || det->isMasked()) {
     return;
   }
 
   const double exp_constant = this->calculateExponential(spectraIndex, det);
   const double scale = this->getProperty("ScaleFactor");
 
-  Mantid::MantidVec &yout = this->outputWS->dataY(spectraIndex);
-  Mantid::MantidVec &eout = this->outputWS->dataE(spectraIndex);
-  // Need the original values so this is not a reference
-  const Mantid::MantidVec yValues = this->inputWS->readY(spectraIndex);
-  const Mantid::MantidVec eValues = this->inputWS->readE(spectraIndex);
-
-  auto yinItr = yValues.cbegin();
-  auto einItr = eValues.cbegin();
-  auto xItr = this->inputWS->readX(spectraIndex).cbegin();
-  auto youtItr = yout.begin();
-  auto eoutItr = eout.begin();
-
-  for (; youtItr != yout.end(); ++youtItr, ++eoutItr) {
-    const double wavelength = (*xItr + *(xItr + 1)) / 2.0;
-    const double effcorr =
-        this->detectorEfficiency(exp_constant * wavelength, scale);
-    *youtItr = (*yinItr) * effcorr;
-    *eoutItr = (*einItr) * effcorr;
-    ++yinItr;
-    ++einItr;
-    ++xItr;
+  const auto &yValues = m_inputWS->y(spectraIndex);
+  const auto &eValues = m_inputWS->e(spectraIndex);
+  auto yInItr = yValues.cbegin();
+  auto eInItr = eValues.cbegin();
+
+  auto &yOut = m_outputWS->mutableY(spectraIndex);
+  auto &eOut = m_outputWS->mutableE(spectraIndex);
+  auto yOutItr = yOut.begin();
+  auto eOutItr = eOut.begin();
+
+  const auto wavelength = m_inputWS->points(spectraIndex);
+  auto wavelenItr = wavelength.cbegin();
+
+  for (; yOutItr != yOut.end(); ++yOutItr, ++eOutItr) {
```

Comment:
maybe use <algorithm> for these. Would have the added benefit of making the code more readable. Would be interested to see if there is an unacceptable performance hit when this is split out into to "loops" for Y and E.